fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/db2/Db2ResourceReferenceDAO.java

```diff
@@ -158,6 +158,13 @@ protected void insertResourceTokenRefs(String resourceType, Collection<ResourceT
                 } else {
                     ps.setNull(4, Types.INTEGER);
                 }
+
+                // compositeId can be null
+                if (xr.getCompositeId() != null) {
+                    ps.setInt(5, xr.getCompositeId());
+                } else {
+                    ps.setNull(5, Types.INTEGER);
+                }
```

Comment:
again --- repeated code, not sure how else we would reconcile.

Reply:
What we really need is a decent sql statement abstraction which we start to use everywhere. I looked a while back to see if there was anything around already, but the ones I saw were too closely couple to other components and would end up bringing in loads of needless dependencies. I started cooking our own in the database-utils project, but it needs some love to make it useful.